build.assets/Makefile

```diff
@@ -20,7 +20,7 @@ OS ?= linux
 ARCH ?= amd64
 
 # Sync with devbox.json.
-GOLANG_VERSION ?= go1.20.7
+GOLANG_VERSION ?= go1.21.0
```

Comment:
Should we update devbox.json too?

Reply:
I doubt devbox has the new version, but I can double-check.

Reply:
Can confirm that it does not yet. The most recent RC is in nixpkgs staging: https://github.com/NixOS/nixpkgs/blob/staging/pkgs/development/compilers/go/1.21.nix.

We could potentially point to that when it's updated fully.